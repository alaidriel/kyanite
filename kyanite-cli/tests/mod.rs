use kyanite::Program;

macro_rules! assert_output {
    ($($path:expr => $name:ident),*) => {
        $(
            #[test]
            fn $name() -> Result<(), Box<dyn std::error::Error>> {
                let mut output = vec![];
                kyanite_cli::run(Program::from_file($path), &mut output)?;
                let output = String::from_utf8(output)?;
                insta::with_settings!({snapshot_path => "../snapshots"}, {
                    let lines: Vec<&str> = output.lines().skip(3).collect();
                    insta::assert_yaml_snapshot!(lines);
                });
                Ok(())
            }
        )*
    }
}

assert_output! {
    "../examples/hello.kya" => hello,
    "../examples/exprs.kya" => exprs,
    "../examples/decls.kya" => decls,
    "../examples/functions.kya" => functions
}
