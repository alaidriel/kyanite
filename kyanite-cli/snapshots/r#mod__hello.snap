---
source: kyanite-cli/tests/mod.rs
expression: "String::from_utf8(output).unwrap().lines().last().unwrap()"
---
Hello, world!
